backend/geneaprove/views/queries.py

```diff
@@ -172,30 +184,35 @@ def get_ancestors(cls, person_id, max_depth=None, skip=0):
         assert isinstance(person_id, int)
 
         with django.db.connection.cursor() as cur:
-            initial = f"VALUES({person_id},0)"
+            # ??? group_concat doesn't exist in postgres
+            if 'postgresql' in database_in_use:
+                initial = f"VALUES({person_id}::bigint, 0::bigint)"
+                group_concat = "string_agg(parents.parent::text, ',') AS parents "
+            else:
+                initial = f"VALUES({person_id},0)"
+                group_concat = "group_concat(parents.parent) AS parents "
```

Comment:
Can we instead make a migration and declare the group_concat aggregate in postgresql (that had been my initial plan, at least) ? "CREATE AGGREGATE group_concat()". I must say I never quite remember the syntax.
Something along the untested:
```
CREATE OR REPLACE FUNCTION comma_concat(text, text) RETURNS text
    LANGUAGE plpgsql
    AS $_$
begin
 if $1 = '' then
  return $2;
 else
  return  $1 || ',' || $2;
 end if;
end;$_$;

CREATE AGGREGATE group_concat(text) (
   SFUNC = textcat, STYPE = text, INITCOND = '');
CREATE AGGREGATE br_concat(text) (
        SFUNC = comma_concat, STYPE = text, INITCOND = '');
```

Then we can avoid the test for postgresql in the code

Reply:
I'm generally averse to migrations unless models etc change in code, as they can eventually grow into a management nightmare. That said, perhaps that's the way to go. There's some discussion of the problem here: https://stackoverflow.com/questions/47637652/can-we-define-a-group-concat-function-in-postgresql

Ideally there would be a portable equivalent approach that is database agnostic, but for now this might be the way to go.

Reply:
I was thinking about it some more, and perhaps I should remove the aggregate altogether, do it in python